odoo/models.py

```diff
@@ -1843,6 +1855,10 @@ def _read_group_format_result(self, data, annotated_groupbys, groupby, domain):
                 if ftype == 'many2one':
                     value = value[0]
                 elif ftype in ('date', 'datetime'):
+                    if ':week_sunday' in gb['groupby']:
```

Comment:
`':week_sunday'` or `'week_sunday'`?

Reply:
I add the colon to avoid cases when field name itself contains ``week_sunday``